functions.py

```diff
@@ -0,0 +1,162 @@
+            fileout = csv.writer(csvFile, delimiter=',', quoting=csv.QUOTE_ALL)
+            fileout.writerow([new_employee.employee_id, new_employee.name, new_employee.phone, new_employee.age])
+
+    def add_employees_from_file(self):
+        """
+        This function gets from the user a file path that contains the data of the employees and adds it to the employees file
+        only if all the data of all employees is supplied.
+        new_employees = a string.
+        """
+        new_employees_file_path = raw_input("Please add the full file path of the employees you wish to add: ")
+        try:
+            with open(new_employees_file_path) as infile:
+                f = open(new_employees_file_path, 'r+')
+                data = f.read()
+                f.close()
+                f = open("employees.csv", "a")
+                f.write(data)
+                f.close()
+        except FileNotFoundError:
+            return "File does not exist."
+
+
+
+    #how to check if all data is supplied in the file
+
+    def delete_employee(self):
+        """
+        This function gets from the user a name of an employee he wishes to remove, writes all the rest of the employees to a temporary
+        file and then deletes the old employees file. Afterwards, it saves it as the main employees file (employees).
+        employees_name = a string.
+        """
+        self.name = exceptions_tests.name()
+        with open("employees_edit.csv", "w") as my_empty_csv:
+            pass
+        with open('employees.csv', 'r') as inp, open('employees_edit.csv', 'w') as out:
+            writer = csv.writer(out)
+            for row in csv.reader(inp):
+                if row[1] != self.name:
+                    writer.writerow(row)
+                else:
+                    return "Sorry, that employee does not exist in this file."
+
+        os.remove('employees.csv')
+        os.rename('employees_edit.csv', 'employees.csv')
+    # IndexError: list index out of range
+    # handle exception if employee does not exist
+
+    def delete_employees_from_file(self):
+        """
+        This function gets from the user a file path that contains the data of the employees and removes the employees from the employees file
+        only if all the data of all employees is supplied.
+        employees_to_delete = a string.
+        """
+        employees_to_delete_file_path = raw_input("Please add the file path of the employees you wish to add: ")
+        try:
+            with open(employees_to_delete_file_path, "r") as f:
+                data_to_erase = f.readlines()  # read data line by line
+            with open("employees.csv", "r") as f:
+                data_to_keep = f.readlines()  # read data line by line
+
+                # open file in write mode
+            with open("employees.csv", "w") as f:
+                for line in data_to_keep:
+                    if line in data_to_erase:
+                        pass
```

Comment:
Why there is pass in here?

Reply:
data_to_keep is the data in the employee's file and. data_to_erase is the data we want to delete. I rewrite the employee's file by comparing between the two. for each line in the employee's file we check if it exists in the data_to_erase file - if so we pass it and not rewrite it to the employees file.

Reply:
You can't use 'pass' in the code, we are using 'pass' while we are writing a code, for our convenience, but we don't keep it when we finish to write the code. -> 
if line not in data_to_erase:
    f.write(line)

Reply:
okay, understood and fixed :)